### PR TITLE
Add roomserver tests (1/?)

### PR DESCRIPTION
**Pull request by S7evinK**
First part of the table tests for the roomserver
- Issue found in the `events` table where the [EventReferenceSHA256 is not set](https://github.com/matrix-org/dendrite/commit/8ae34c4e5f78584f0edb479f5a893556d2b95d19) correctly
- Minor changes - moved the variable declaration outside `rows.Next()`, which slightly reduces allocations

### CODE REVIEW

File: roomserver/storage/tables/event_types_table_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				ctx, nil, eventType,
+			); err != nil {
+				t.Fatalf("unable to insert eventJSON: %s", err)
+			}
+			eventTypeMap[eventType] = eventTypeNID
+			gotEventTypeNID, err = tab.SelectEventTypeNID(ctx, nil, eventType)
+			assert.NoError(t, err)
+			assert.Equal(t, eventTypeNID, gotEventTypeNID)
+		}
+		eventTypeNIDs, err := tab.BulkSelectEventTypeNID(ctx, nil, []string{"dummyEventType0", "dummyEventType3"})
```
Comment (kegsay) [inline roomserver/storage/tables/event_types_table_test.go:62]: Also please test the negative case (a NID which is not in the DB).

File: roomserver/storage/tables/event_types_table_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		defer close()
+		ctx := context.Background()
+		var eventTypeNID, gotEventTypeNID types.EventTypeNID
+		var err error
+		// create some dummy data
+		eventTypeMap := make(map[string]types.EventTypeNID)
+		for i := 0; i < 10; i++ {
+			eventType := fmt.Sprintf("dummyEventType%d", i)
+			if eventTypeNID, err = tab.InsertEventTypeNID(
+				ctx, nil, eventType,
```
Comment (kegsay) [inline roomserver/storage/tables/event_types_table_test.go:53]: Also test duplicate inserts of the same event type please.

File: roomserver/storage/tables/event_state_keys_table_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		err = sqlite3.CreateEventStateKeysTable(db)
+		assert.NoError(t, err)
+		tab, err = sqlite3.PrepareEventStateKeysTable(db)
+	}
+	assert.NoError(t, err)
+
+	return tab, close
+}
+
+func Test_EventStateKeysTable(t *testing.T) {
```
Comment (kegsay) [inline roomserver/storage/tables/event_state_keys_table_test.go:41]: See comments on event type table test

File: roomserver/storage/tables/event_json_table_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		// create some dummy data
+		for i := 0; i < 10; i++ {
+			err := tab.InsertEventJSON(
+				context.Background(), nil, types.EventNID(i),
+				[]byte(fmt.Sprintf(`{"value":%d"}`, i)),
+			)
+			assert.NoError(t, err)
+		}
+		// select a subset of the data
+		values, err := tab.BulkSelectEventJSON(context.Background(), nil, []types.EventNID{1, 2, 3, 4, 5})
```
Comment (kegsay) [inline roomserver/storage/tables/event_json_table_test.go:54]: Also test:
 - No matching events (invalid nids)
 - Mixed matches/no matches
 - Single NID

**Review by kegsay — APPROVED** [review]
